src/main/java/org/madhuri/app/message/handler/LoginHandler.java

```diff
@@ -0,0 +1,43 @@
+package org.madhuri.app.message.handler;
+
+import java.util.Optional;
+
+import org.madhuri.app.message.dao.ProfileDAO;
+import org.madhuri.app.message.model.Profile;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+@Path("/login")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+
+public class LoginHandler {
+
+    private ProfileDAO profileDAO;
+
+    public LoginHandler() {
+        this.profileDAO = new ProfileDAO();
+    }
+
+    @POST
+    public Response login(Profile profile) {
+        Optional<Profile> foundProfile = profileDAO.findByUsername(profile.getUsername());
+
+        if (foundProfile.isPresent()) {
+            if (foundProfile.get().getPassword().equals(profile.getPassword())) {
```

Comment:
you are already checking password in login service, why are you using profile DAO in login handler directly ? 
you can use login service right ?